### ISSUE TITLE
Failure to resolve leafref of identityref type

### ISSUE THREAD

**Issue by abhikeshav** [CESNET/libyang#174]
Given below yang,

```
module atypes {
    namespace "urn:atypes";
    prefix atypes;
    identity ltype {
    }
    identity ltype1 {
        base ltype;
    }
}
```

```
module a {
...
list x {
   key "l";
        leaf l {
            type leafref {
                path "../config/l";
            }
        }
        container config {
            leaf l {
                type identityref {
                    base atypes:ltype;
                }
            }
        }
   }
}
```

Below XML fails to be parsed with below error:

```
<x xmlns="urn:a" xmlns:idx="urn:atypes">
  <l>idx:ltypes1</l>
  <config>
    <l>idx:ltypes1</l>
  </config>
</x>
```

```
libyang[0]: Failed to resolve identityref "idx:ltypes1". (path: /a:x[l='idx:ltypes1']/l)
```

**Comment by rkrejci** [comment]
which is correct because module `atypes` defines identity `ltype1`, not `ltypes1` used in the data.

**Comment by abhikeshav** [comment]
Sorry about that. I am seeing the failure even with using the correct XML tag in the data:

```
<x xmlns="urn:a" xmlns:idx="urn:atypes">
  <l>idx:ltype1</l>
  <config>
    <l>idx:ltype1</l>
  </config>
</x>
```

**Comment by abhikeshav** [comment]
FYI.. I'm still seeing this issue on the devel branch as well. Thanks

**Comment by rkrejci** [comment]
should be fixed in devel branch (0.11.36)
